### PR TITLE
Added some basic logging for read operations

### LINKED ISSUE #174: [New Feature] Adding query costs to debug logging to aid developer inner loop
**Is your feature request related to a problem? Please describe.**
Currently there is limited support to determine how many RUs are used for each query. This is currently only surfaced in the paging functionality.

**Describe the solution you'd like**
After discussions on twitter with @mumby0168 it was determined that we could add the costs to debug statements to allow for this to aid developers locally - https://twitter.com/billydev5/status/1482287510990426113

### PR DESCRIPTION
**Pull request by mumby0168**
Added some basic logging following the [high-performance logging guidance](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/loggermessage?view=aspnetcore-6.0).

Closes #174 

I still think we can do a little more of this for some of the create & update operations.

I plan to add a docs page for this @IEvangelist if you approve of this 😊

### CODE REVIEW

**Review by WestDiscGolf — COMMENTED** [review]
Looks good to me 😄

File: src/Microsoft.Azure.CosmosRepository/Logging/EventIds.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        public static readonly EventId CosmosPointReadStarted = new(
+            15_101,
+            nameof(CosmosPointReadStarted));
+
+        public static readonly EventId CosmosPointReadExecuted = new(
+            15_102,
+            nameof(CosmosPointReadExecuted));
+
+        public static readonly EventId CosmosQueryConstructed = new(
+            15_102,
```
Comment (mumby0168) [inline src/Microsoft.Azure.CosmosRepository/Logging/EventIds.cs:27]:
```suggestion
            15_103,
```

File: src/Microsoft.Azure.CosmosRepository/Logging/EventIds.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            nameof(CosmosQueryConstructed));
+
+        public static readonly EventId CosmosQueryExecuted = new(
+            15_103,
```
Comment (mumby0168) [inline src/Microsoft.Azure.CosmosRepository/Logging/EventIds.cs:31]:
```suggestion
            15_104,
```

File: src/Microsoft.Azure.CosmosRepository/Logging/EventIds.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace Microsoft.Azure.CosmosRepository.Logging
+{
+    internal static class EventIds
+    {
+        //15_000 - 15_100 Debug Events
+        public static readonly EventId CosmosItemRead = new(
+            15_001,
+            nameof(CosmosItemRead)
+        );
```
Comment (mumby0168) [inline src/Microsoft.Azure.CosmosRepository/Logging/EventIds.cs:15]:
```suggestion
            nameof(CosmosItemRead));
```

File: src/Microsoft.Azure.CosmosRepository/Logging/LoggerExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+            LoggerMessageDefinitions.PointReadStarted(logger, typeof(TItem).Name, id, partitionKey, null);
+
+        public static void LogPointReadExecuted<TItem>(this ILogger logger, double ruCharge) where TItem : IItem =>
+            LoggerMessageDefinitions.PointReadExecuted(logger, typeof(TItem).Name, ruCharge, null);
+
+        public static void LogQueryConstructed<TItem>(this ILogger logger, IQueryable<TItem> queryable) where TItem : IItem =>
+            LoggerMessageDefinitions.QueryConstructed(logger, typeof(TItem).Name, queryable.ToString(), null);
+
+        public static void LogQueryExecuted<TItem>(this ILogger logger, IQueryable<TItem> queryable, double charge) where TItem : IItem =>
+            LoggerMessageDefinitions.QueryExecuted(logger, typeof(TItem).Name, charge, queryable.ToString(), null);
```
Comment (IEvangelist) [inline src/Microsoft.Azure.CosmosRepository/Logging/LoggerExtensions.cs:33]:
```suggestion
        public static void LogPointReadExecuted<TItem>(
            this ILogger logger,
            double ruCharge) where TItem : IItem =>
            LoggerMessageDefinitions.PointReadExecuted(logger, typeof(TItem).Name, ruCharge, null);

        public static void LogQueryConstructed<TItem>(
            this ILogger logger,
            IQueryable<TItem> queryable) where TItem : IItem =>
            LoggerMessageDefinitions.QueryConstructed(logger, typeof(TItem).Name, queryable.ToString(), null);

        public static void LogQueryExecuted<TItem>(
            this ILogger logger,
            IQueryable<TItem> queryable,
            double charge) where TItem : IItem =>
            LoggerMessageDefinitions.QueryExecuted(logger, typeof(TItem).Name, charge, queryable.ToString(), null);
```

**Review by IEvangelist — APPROVED** [review]
One nit

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**